spec/controllers/events_controller_spec.rb

```diff
@@ -19,12 +19,35 @@
 # that an instance is receiving a specific message.
 
 RSpec.describe EventsController, type: :controller do
+  let(:date1) { Date.today }
+  let(:date2) { Date.today.next_day }
+  let(:date3) { Date.today.next_day(2) }
+  let(:date4) { Date.today.next_day(2) }
+
+  # this is the format expected by our controller due to reasons outlined
+  # in EventsController#date_range_params
+  let(:valid_attributes_post) {
+    {
+        event: {
+          name: 'Test',
+          description: 'Test',
+          max_participants: 1,
+          active: false,
+        },
+        date_ranges: {
+          start_date: [
```

Comment:
You could use a map function for better readability, but for this test it should be acceptable